generate.sh

```diff
@@ -5,7 +5,28 @@ if [ $# -eq 0 ]; then
     exit 1
 fi
 
-if command -v podman > /dev/null
+
+get_container_engine () {
+
+    # allow override from callers such as oci-env
+    if [ -z ${COMPOSE_BINARY} ]; then
```

Comment:
This doesn't make sense to me. Here you know it's empty, so what's the point? Did you mean:
```suggestion
    if [[ -n ${COMPOSE_BINARY} ]]; then
```

Note I'm also suggesting to use the bash double brackets. That properly deals with it not being present. Otherwise it may end up evaluating to `if [ -n  ]; then` which is a syntax error. The alternative is to quote the variable.